discord/poll.py

```diff
@@ -0,0 +1,577 @@
+            return
+
+        # The message's poll contains the more up to date data.
+        self._expiry = message.poll.expires_at
+        self._finalized = message.poll._finalized
+
+    def _update_results(self, data: PollResultPayload) -> None:
+        self._finalized = data['is_finalized']
+
+        for count in data['answer_counts']:
+            answer = self.get_answer(int(count['id']))
+            if not answer:
+                continue
+
+            answer._update_with_results(count)
+
+    def _handle_vote(self, answer_id: int, added: bool, self_voted: bool = False):
+        answer = self.get_answer(answer_id)
+        if not answer:
+            return
+
+        answer._handle_vote_event(added, self_voted)
+
+    @classmethod
+    def _from_data(cls, *, data: PollPayload, message: Message, state: ConnectionState) -> Self:
+        multiselect = data.get('allow_multiselect', False)
+        layout_type = try_enum(PollLayoutType, data.get('layout_type', 1))
+        question_data = data.get('question')
+        question = question_data.get('text')
+        expiry = utils.parse_time(data['expiry'])  # If obtained via API, then expiry is set.
+        duration = expiry - message.created_at
+        # self.created_at = message.created_at
+        # duration = self.created_at - expiry
+
+        if (duration.total_seconds() / 3600) > 168:  # As the duration may exceed little milliseconds then we fix it
+            duration = datetime.timedelta(days=7)
+
+        self = cls(
+            duration=duration,
+            multiple=multiselect,
+            layout_type=layout_type,
+            question=question,
+        )
+        self._answers = {
+            int(answer['answer_id']): PollAnswer(data=answer, message=message, poll=self) for answer in data['answers']
+        }
+        self._message = message
+        self._state = state
+        self._expiry = expiry
+
+        try:
+            self._update_results(data['results'])
+        except KeyError:
+            pass
+
+        return self
+
+    def _to_dict(self) -> PollCreatePayload:
+        data: PollCreatePayload = {
+            'allow_multiselect': self.multiple,
+            'question': self._question_media.to_dict(),
+            'duration': self.duration.total_seconds() / 3600,
+            'layout_type': self.layout_type.value,
+            'answers': [answer._to_dict() for answer in self.answers],
+        }
+        return data
+
+    def __repr__(self) -> str:
+        return f"<Poll duration={self.duration} question=\"{self.question}\" answers={self.answers}>"
+
+    @property
+    def question(self) -> str:
+        """:class:`str`: Returns this poll answer question string."""
+        return self._question_media.text
+
+    @property
+    def emoji(self) -> Optional[PartialEmoji]:
+        """Optional[:class:`PartialEmoji`]: Returns the emoji for this poll's question."""
+
+        return None  # As of now, polls questions don't support emojis
```

Comment:
```suggestion
```

I don't see the point of this.

Reply:
Yeah, I thought of it in the last day, but just couldn't find the words to say it. Anyways, it was removed in the latest commits.